concordium-consensus/src/Concordium/GlobalState.hs

```diff
@@ -583,20 +589,43 @@ instance GlobalStateConfig DiskTreeDiskBlockConfig where
 
     initialiseExistingGlobalState _ DTDBConfig{..} = do
       -- check if all the necessary database files exist
-      existingDB <- checkExistingDatabase dtdbTreeStateDirectory dtdbBlockStateFile
-      pbscAccountCache <- liftIO $ Accounts.newAccountCache (rpAccountsCacheSize dtdbRuntimeParameters)
-      pbscModuleCache <- liftIO $ Modules.newModuleCache (rpModulesCacheSize dtdbRuntimeParameters)
-      if existingDB then do
-        pbscBlobStore <- liftIO $ do
-          -- the block state file exists, is readable and writable
-          -- we ignore the given block state parameter in such a case.
-          loadBlobStore dtdbBlockStateFile
-        let pbsc = PersistentBlockStateContext{..}
-        logm <- ask
-        skovData <- liftIO (runLoggerT (loadSkovPersistentData dtdbRuntimeParameters dtdbTreeStateDirectory pbsc) logm
-                    `onException` (closeBlobStore pbscBlobStore))
-        return (Just (pbsc, skovData))
-      else return Nothing
+      let treeStateFile = dtdbTreeStateDirectory </> "data.mdb"
+      bsPathEx <- liftIO $ doesPathExist dtdbBlockStateFile
+      tsPathEx <- liftIO $ doesPathExist treeStateFile
+      if | bsPathEx && tsPathEx -> do
+             checkRWFile dtdbBlockStateFile BlockStatePermissionError
+             checkRWFile treeStateFile TreeStatePermissionError
+             mapM_ (logEvent TreeState LLTrace) ["Existing database found.", "TreeState filepath: " ++ show dtdbBlockStateFile, "BlockState filepath: " ++ show treeStateFile]
+             logm <- ask
+             liftIO $ do
+               pbscAccountCache <- Accounts.newAccountCache (rpAccountsCacheSize dtdbRuntimeParameters)
+               pbscModuleCache <- Modules.newModuleCache (rpModulesCacheSize dtdbRuntimeParameters)
+               pbscBlobStore <- loadBlobStore dtdbBlockStateFile
+               let pbsc = PersistentBlockStateContext{..}
+               skovData <- runLoggerT (loadSkovPersistentData dtdbRuntimeParameters dtdbTreeStateDirectory pbsc) logm
+                                   `onException` closeBlobStore pbscBlobStore
+               return (Just (pbsc, skovData))
+         | bsPathEx -> do
+             logEvent GlobalState LLWarning "Block state file exists, but tree state file does not. Deleting the block state file."
+             liftIO $ removeFile dtdbBlockStateFile
+             return Nothing
+         | tsPathEx -> do
+             logEvent GlobalState LLWarning "Tree state file exists, but block state file does not. Deleting the tree state file."
```

Comment:
TreeState is not a file. I recommend that the message talks about "database" instead.